bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,181 @@
+
+# Defining all words in the list of words as a global variable
+with Path("bot/resources/fun/hangman_words.txt").resolve().open(mode="r", encoding="utf-8") as f:
+    ALL_WORDS = [line.strip('\n') for line in f.readlines()]
+
+# Defining a list of images that will be used for the game to represent the hangman person
+IMAGES = {
+    6: "https://cdn.discordapp.com/attachments/859123972884922418/887067234760032296/hangman0.png",
+    5: "https://cdn.discordapp.com/attachments/859123972884922418/887067240204206220/hangman1.png",
+    4: "https://cdn.discordapp.com/attachments/859123972884922418/887067244578881626/hangman2.png",
+    3: "https://cdn.discordapp.com/attachments/859123972884922418/887067248546685029/hangman3.png",
+    2: "https://cdn.discordapp.com/attachments/859123972884922418/887067252099268608/hangman4.png",
+    1: "https://cdn.discordapp.com/attachments/859123972884922418/887067255429554197/hangman5.png",
+    0: "https://cdn.discordapp.com/attachments/859123972884922418/887067260726939728/hangman6.png",
+}
+
+
+class Hangman(commands.Cog):
+    """
+    Cog for the Hangman game.
+
+    Hangman is a classic game where the user tries to guess a word, with a limited amount of tries.
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command()
+    async def hangman(
+            self,
+            ctx: commands.Context,
+            min_length: int = 0,
+            max_length: int = 25,
+            min_unique_letters: int = 0,
+            max_unique_letters: int = 25,
+            singleplayer: Literal["s", "m", "S", "M"] = "s",
+    ) -> None:
+        """
+        Play hangman against the bot, where you have to guess the word it has provided!
+
+        The arguments for this command mean:
+        - min_length: the minimum length you want the word to be (i.e. 2)
+        - max_length: the maximum length you want the word to be (i.e. 5)
+        - min_unique_letters: the minimum unique letters you want the word to have (i.e. 4)
+        - max_unique_letters: the maximum unique letters you want the word to have (i.e. 7)
+        - singleplayer: writing 's' means you want to play by yourself, and only you can suggest letters,
+            - writing 'm' means you want multiple players to join in and guess the word.
+        """
+        # Changing singleplayer to a boolean
+        singleplayer = True if singleplayer.lower() == 's' else False
+
+        # Filtering the list of all words depending on the configuration
+        filtered_words = [
+            word for word in ALL_WORDS
+            if min_length < len(word) < max_length
+            and min_unique_letters < len(set(word)) < max_unique_letters
+        ]
+
+        if not filtered_words:
+            filter_not_found_embed = Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="No words could be found that fit all filters specified.",
+                color=Colours.soft_red,
+            )
```

Comment:
This embed could be created in advance, but the performance impact is rather negligible, so this is more of an idea, rather than a suggestion.

Reply:
There are other embeds below, and i dont wanna repeat myself each time, so think of this comment applying to them as well.

Reply:
That would not change the title each time

Reply:
Fair Point